src/Compile-SourceScript/Public/Compile-SourceScript.ps1

```diff
@@ -123,33 +128,60 @@ function Compile-SourceScript {
 
-            # Prepare for compilation
-            "Compiling..." | Write-Host -ForegroundColor Cyan
+            # Generate command line arguments
             $epoch = [Math]::Floor([decimal](Get-Date(Get-Date).ToUniversalTime()-uformat "%s"))
-            $stdInFile = New-Item -Path (Join-Path $SCRIPTING_DIR ".$epoch") -ItemType File -Force
-            '1' | Out-File -FilePath $stdInFile.FullName -Force -Encoding utf8
+            $stdInFile = Join-Path $SCRIPTING_DIR ".$epoch"
+            $tempDir = Join-Path ([System.IO.Path]::GetTempPath()) (New-Guid).Guid
+            $stdoutFile = Join-Path $tempDir 'stdout'
+            $stderrFile = Join-Path $tempDir 'stderr'
             $processArgs = @{
                 FilePath = $compiler.FullName
+                ArgumentList = @(
+                    if ($PSBoundParameters['SkipWrapper']) {
+                        $sourceFile.Name
+                        "-o$($MOD[$MOD_NAME]['compiled_dir_name'])/$($sourceFile.Basename)$($MOD[$MOD_NAME]['plugin_ext'])"
+                    }else {
+                        $sourceFile.Name
+                    }
+                )
                 WorkingDirectory = $SCRIPTING_DIR
-                RedirectStandardInput = $stdInFile.FullName
+                RedirectStandardInput = $stdInFile
+                RedirectStandardOutput = $stdoutFile
+                RedirectStandardError = $stderrFile
                 Wait = $true
                 NoNewWindow = $true
+                PassThru = $true
             }
-            if ($PSBoundParameters['SkipWrapper']) {
-                $processArgs['ArgumentList'] = @(
-                    $sourceFile.Name
-                    "-o$($MOD[$MOD_NAME]['compiled_dir_name'])/$($sourceFile.Basename)$($MOD[$MOD_NAME]['plugin_ext'])"
-                )
-            }else {
-                $processArgs['ArgumentList'] = @(
-                    $sourceFile.Name
-                )
+
+            # Prepare compilation environment
+            if ($item = New-Item -Path $stdInFile -ItemType File -Force) {
+                # This dummy input bypasses the 'Press any key to continue' prompt of the compiler
+                '1' | Out-File -FilePath $item.FullName -Force -Encoding utf8
             }
+            New-Item $tempDir -ItemType Directory -Force > $null
             New-Item -Path $COMPILED_DIR -ItemType Directory -Force | Out-Null
 
             # Begin compilation
+            "Compiling..." | Write-Host -ForegroundColor Cyan
             if ($PSBoundParameters['SkipWrapper']) { "Compiling $($sourceFile.Name)..." | Write-Host -ForegroundColor Yellow }
-            Start-Process @processArgs
+
+            # Compile
+            $global:LASTEXITCODE = 0
+            $p = Start-Process @processArgs
+            $stdout = Get-Content $stdoutFile
+            $stdout | Write-Host
+            $stderr = Get-Content $stderrFile
+            foreach ($line in $stdout) {
```

Comment:
Is there a need for outputting stderr?

Reply:
As we said, `stderr` will be empty. No harm in outputting, though if it were not, because `stdout` and `stderr` are separated, the output will no longer be in the original sequence if printing `stderr` after printing `stdout`.

Reply:
We could add it in then.

Reply:
Should we take the risk to output stderr or stay confident about just outputting stdout only?

Reply:
i dont see any risk with outputting `stderr`, except the scenario where the lack of continuity in the output does not match the developer's expectation. 

A benefit in outputting `stdout` is in the case where there are missing messages in `stdout`, with `stderr` a developer might be able to at least be able to have some sort of output to fix any issues which cause the compile to spit out messages on `stderr` which we currently have no knowledge of. 

@joeltimothyoh

Reply:
Maybe you could commit the changes you think makes most sense so we can close this PR? @leojonathanoh

Reply:
fixed